filter/ngx_http_zstd_filter_module.c

```diff
@@ -215,12 +215,7 @@ ngx_http_zstd_header_filter(ngx_http_request_t *r)
 
     r->gzip_vary = 1;
 
-    if (!r->gzip_tested) {
```

Comment:
As per my analysis, the orignal code won't enable zstd compression when `Accept-Encoding` is not `zstd`.

Reply:
My compression related nginx settings over all locations:

```nginx
    gzip on;
    gzip_static on;
    gzip_vary on;
    gzip_comp_level 5;
    gzip_min_length 256;
    gzip_proxied expired no-cache no-store private auth;
    gzip_types application/atom+xml application/geo+json application/json application/ld+json application/manifest+json application/rdf+xml application/vnd.ms-fontobject application/wasm application/x-rss+xml application/x-web-app-manifest+json application/xhtml+xml application/xliff+xml application/xml font/collection font/otf font/ttf image/bmp image/svg+xml image/vnd.microsoft.icon text/cache-manifest text/calendar text/css text/csv text/html text/javascript text/markdown text/plain text/vcard text/vnd.rim.location.xloc text/vtt text/x-component text/xml;
    zstd on;
    zstd_comp_level 9;
    zstd_min_length 256;
    zstd_static on;
    zstd_types application/atom+xml application/geo+json application/json application/ld+json application/manifest+json application/rdf+xml application/vnd.ms-fontobject application/wasm application/x-rss+xml application/x-web-app-manifest+json application/xhtml+xml application/xliff+xml application/xml font/collection font/otf font/ttf image/bmp image/svg+xml image/vnd.microsoft.icon text/cache-manifest text/calendar text/css text/csv text/html text/javascript text/markdown text/plain text/vcard text/vnd.rim.location.xloc text/vtt text/x-component text/xml;
```

Before this patch this happened:

```
 ▶ curl https://rss-bridge.supersandro.de/static/style.css -H"Accept-encoding: gzip" -o/dev/null -v |& rg encoding
* h2h3 [accept-encoding: gzip]
> accept-encoding: gzip
< vary: Accept-Encoding
< content-encoding: zstd

 ▶ curl https://rss-bridge.supersandro.de/static/style.css -H"Accept-encoding: zstd" -o/dev/null -v |& rg encoding
* h2h3 [accept-encoding: zstd]
> accept-encoding: zstd
< vary: Accept-Encoding
```

and with this patch I do not get gzip returned when zstd is enabled.


I did some more debugging and narrowed it down to the following setting causing this: ``gzip_static on;`` When I disable gzip_static or apply the patch, zstd is not longer returned when gzip is requested.

Reply:
Got your mean. Thanks.